Replicator/ReplicatorOptions.hh

```diff
@@ -36,20 +39,24 @@ namespace litecore { namespace repl {
         PropertyDecryptor       propertyDecryptor       {nullptr};
         void*                   callbackContext         {nullptr};
         std::atomic<C4ReplicatorProgressLevel> progressLevel {kC4ReplProgressOverall};
-        bool                    collectionAware         {true};
-
+        mutable bool            collectionAware         {true};
+        mutable bool            isActive                {true};
```

Comment:
I also think this can be a method rather than an item.

Reply:
Another option is to add isActive() and isCollectionAware() in the Replicator class. If doing that, isCollectionAware() can be computed for both Active and Passive replicator.

Reply:
Current architecture is that, for a particular replication work, there are a group of workers, headed by Replicator. All the workers in that group share the same Options object. In child workers, calling replicator()->isActive() is less safe than _options->isActive() because replicator() returns a bare pointer whereas _options is shared ponter.

Reply:
I submitted my motivation of having them in _options in another response.

Reply:
Changed them to methods